### ISSUE TITLE
Typing gets laggy over time on macOS

### ISSUE THREAD

**Issue by wch** [rstudio/rstudio#8442]
This is similar to #1539, but not quite the same.

At first when I start up the IDE, it's reasonably snappy, but if I keep it open and use it for days, it becomes laggy -- I'm guessing around .25s between pressing the key and having it show up on screen. 

Similar to #1539, shrinking the window makes it more responsive. Right-clicking and clicking on Reload doesn't help.

    RStudio Edition : desktop
    RStudio Version : 1.4.1016
    OS Version      : macOS Catalina 10.15.5
    R Version       : 4.0.3

**Comment by wch** [comment]
From some experimentation I did with the Chrome profiler:

I checked “Enable advanced paint instrumentation (slow)“, and it looks like it spends almost 90% of the time in “Composite Layers”.

![image](https://user-images.githubusercontent.com/86978/99838594-027aed80-2b2f-11eb-8db7-80a25268bc26.png)

Also, it looks like the cursor DOM element causes the entire editor panel to redraw every time I type something. When I set it to `display: none;`, it doesn’t redraw the whole editor panel, and subjectively it seems faster. (Note that my old session froze, so this is a new session which isn’t as laggy, so it’s hard to tell how big the speedup is.)

I've put a video of this here: https://photos.app.goo.gl/hD9aXSzWEM2p7Urf6

**Comment by wch** [comment]
Another observation: when the typing gets laggy, scrolling also becomes very laggy.

**Comment by wch** [comment]
Keeping this issue alive...

**Comment by kevinushey** [comment]
Marking this as "probably fixed by Electron".

**Comment by wch** [comment]
If I experience the issue with Chrome, will Electron help? I thought Electron also used Chromium.

**Comment by gtritchie** [comment]
@wch Yes, if you are seeing this using latest Chrome for RStudio Server, then not likely to be fixed by Electron.

However if you are seeing this in RStudio Desktop, that has a version of Chromium that is several years old, so it's possible (no idea how probable) it could be improved via Electron, which usually has a very recent Chromium.

**Comment by wch** [comment]
@gtritchie that makes sense. I am using the desktop version.

**Comment by ronblum** [comment]
@wch Would you be willing to try it out with Electron?

**Comment by wch** [comment]
@ronblum The work I've been doing recently mostly involves working outside of the IDE, so I don't think I'm in a good place to give feedback on this issue right now.

**Comment by mikebessuille** [comment]
Closing this on the assumption this is likely fixed by our Electron version.  If we can reproduce it, we'll reopen.
